### PR TITLE
More student progress table improvements.

### PR DESCRIPTION
**Pull request by somiaj**
Remove the pre elements and use a nested table to display the
  problems scores for each set as mentioned in #1616.

  This is done by removing the custom grade_set method and creating
  a new generate_problem_table method, which creates a two row table
  listing the problem status and number of incorrect attempts.

  Switches to using the WeBWorK::Utils::grade_set method to grade
  each set instead of the custom grade_set method.

  Removes the success indicator index which was no longer being
  shown on the student progress table.

  Removes the two instances of using obsolete align attribute
  in the table headers.

### CODE REVIEW

File: lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm [hunk 1]
```diff
@@ -975,225 +909,104 @@ sub displaySets {
...
-			   $twoString,
-			   $totalRight,
-			   $total,
-			   $num_of_attempts,
-			   $num_of_problems
-		);
+	return $problemTable;
 }
 #################################
 # Utility function NOT a method
```
Comment (drgrice1) [inline lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm:1012]: You should delete the `fourSpaceFill` sub below here since it is no longer used.

**Review by drgrice1 — COMMENTED** [review]
Now that I actually see this in the context of the rest of the table and page it doesn't look as good as I thought it would.  It is possible that with some css it could be made to look a little better, but still not the best.  The issue is that the width styles that you added on lines 541 and 1004 don't work that well.  When there are enough problems to make the percent width small the scores (particularly a 100) can exceed the given width, and then the problem scores tables below do not align correctly with the header table above.

I think the only way that alignment will actually work out is to do it the same as on the grades page and on the student progress page for a single user.  It should actually be easier to do that here than on those pages.  Here you have the advantage of knowing the number of problems, and that will be the same for all students.

**Comment by drgrice1** [comment]
Here is a screenshot to show you what I am talking about with the alignment issue.
misaligned

This would get considerably worse if there were more nonzero scores.

**Comment by somiaj** [comment]
The class I was using as a test case only has like 8-12 problems per set, so I overlooked what happens with larger numbers. I think I've done a lot of the setup with removing some of that old code, so I will work on integrating the table vs using nested tables.

**Review by drgrice1 — APPROVED** [review]
This looks much better now.  It is also more consistent in appearance with the other pages that do this sort of thing.

**Comment by somiaj** [comment]
I just caught a issue in computing the number of columns to span when 'no tests taken' is shown and the problems are not visible on a gateway quiz progress. Update fixed that.

**Review by pstaabp — COMMENTED** [review]
This looks good.  Nice improvement.

**Comment by drgrice1** [comment]
I will count @pstaabp's comment as an approval and merge this.

**Pull request merged**